src/Proxy/EventDispatcherProxy.php

```diff
@@ -3,23 +3,47 @@
 namespace Yiisoft\Yii\Debug\Proxy;
 
 use Psr\EventDispatcher\EventDispatcherInterface;
-use Yiisoft\Yii\Debug\Collector\CollectorInterface;
+use Yiisoft\Yii\Debug\Collector\EventCollectorInterface;
+use Yiisoft\Yii\Debug\Debugger;
+use Yiisoft\Yii\Web\Event\ApplicationShutdown;
+use Yiisoft\Yii\Web\Event\ApplicationStartup;
 
 final class EventDispatcherProxy implements EventDispatcherInterface
 {
     private EventDispatcherInterface $dispatcher;
-    private CollectorInterface $collector;
+    private EventDispatcherInterface $debugEventDispatcher;
+    private EventCollectorInterface $collector;
+    private Debugger $debugger;
 
-    public function __construct(EventDispatcherInterface $dispatcher, CollectorInterface $collector)
-    {
+    public function __construct(
+        EventDispatcherInterface $dispatcher,
+        EventDispatcherInterface $debugEventDispatcher,
+        EventCollectorInterface $collector,
+        Debugger $debugger
+    ) {
         $this->dispatcher = $dispatcher;
+        $this->debugEventDispatcher = $debugEventDispatcher;
         $this->collector = $collector;
+        $this->debugger =  $debugger;
     }
 
     public function dispatch(object $event)
     {
         $this->collector->collect($event);
+        $this->processEvent($event);
```

Comment:
`ApplicationStartup` won't catch by collector, because it won't started at this moment...

Reply:
> `ApplicationStartup` won't catch by collector, because it won't started at this moment...
Everything works in the yii-demo app.